downgrade

```diff
@@ -243,24 +274,90 @@ declare -a installed
 cli() {
   while [[ -n "$1" ]]; do
     case "$1" in
+      --pacman)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN="$1"
+        else
+          (
+          gettext "Missing --pacman argument"
+          echo
+          gettext "$(usage)"
+          ) >&2
+          exit 1
+        fi
+        ;;
+      --pacman-conf)
+        if [[ -n "$2" ]]; then
+          shift
+          PACMAN_CONF="$1"
+        else
+          (
+          gettext "Missing --pacman-conf argument"
+          echo
+          gettext "$(usage)"
+          ) >&2
+          exit 1
+        fi
+        ;;
+      --arch)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ARCH="$1"
+        else
+          (
+          gettext "Missing --arch argument"
+          echo
+          gettext "$(usage)"
+          ) >&2
+          exit 1
+        fi
+        ;;
+      --ala-url)
+        if [[ -n "$2" ]]; then
+          shift
+          DOWNGRADE_ALA_URL="$1"
+        else
+          (
+          gettext "Missing --ala-url argument"
+          echo
+          gettext "$(usage)"
+          ) >&2
+          exit 1
+        fi
+        ;;
+      --ala)
+        DOWNGRADE_FROM_ALA=1
+        DOWNGRADE_FROM_CACHE=0
+        ;;
+      --cached)
+        DOWNGRADE_FROM_ALA=0
+        DOWNGRADE_FROM_CACHE=1
+        ;;
+      --nosudo)
+        DOWNGRADE_NOSUDO=1
+        ;;
+      -h | --help)
+        gettext "$(usage)"
+        exit 1
+        ;;
       --)
         shift
         pacman_args=("$@")
         break
         ;;
       *)
         terms+=("$1")
-        shift
         ;;
     esac
+    shift
   done
 
   if ((!"${#terms[@]}")); then
     (
-      gettext 'usage: downgrade <pkg> [pkg...] [-- pacman_option...]'
```

Comment:
Oh, actually, I think this usage was following a defined convention. See https://github.com/pbrisbin/downgrade/issues/69.

I think you want your new usage to be:

```
usage: downgrade [option...] <pkg> [pkg...] [-- pacman_option...]
```

Reply:
Committed a correction

Reply:
Also added two possible syntaxes in the usage, since both are technically possible. Also I see many other CLI tools using alternate syntaxes which I personally find cool.

```
Usage: downgrade [option...] <pkg> [pkg...] [-- pacman_option...]
       downgrade <pkg> [pkg...] [option...] [-- pacman_option...]
```